### ISSUE TITLE
AndroidStudio 'define inline' problem

### ISSUE THREAD

**Issue by T-Nosaka** [libgit2/libgit2#6574]
### Reproduction steps
Include libgit2 in CMake.

I need build in libgit2 by AndroidStudio.
But, below happened on compile error.
Need modify in-line code.
/src/libgit2/util/git2_util.h LINE 96

The error content is "error: unknown type name 'inline'".

Inline seems to be affected when "# include <arpa/inet.h>" in Android compilation environment.
Therefore, it works fine by modifying as follows.

```
#ifdef ANDROID
#define inline
#endif
#include <arpa/inet.h>
```

makebuild.sh
```
#!/bin/bash

export LIBGIT2=libgit2-1.6.4

#TOOLCHAIN
export NDKVER=25.1.8937393

PLATFORMNAME=`uname -s`
if [ $PLATFORMNAME = "Darwin" ]; then
export ANDROID_SDK=~/Library/Android/Sdk
LLVMPLATFORM="darwin-x86_64"
else
export ANDROID_SDK=~/Android/Sdk
LLVMPLATFORM="linux-x86_64"
fi
export NDK=${ANDROID_SDK}/ndk/${NDKVER}
export TOOLCHAIN=$NDK/toolchains/llvm/prebuilt/${LLVMPLATFORM}
export CMAKE=${ANDROID_SDK}/cmake/3.22.1/bin/cmake
export OUTPUT_DIR=`pwd`/output
export OPENSSL_DIR=../openssl/output
export LIBSSH2_DIR=../libssh2/output
export MAKE=${NDK}/prebuilt/${LLVMPLATFORM}/bin/make

#API
export API_LEVEL=26

make_android() {
  NDKARCH=$1

  ${CMAKE}  \
  -H${LIBGIT2} \
  -Bmakegenerate/${NDKARCH} \
  -DOPENSSL_INCLUDE_DIR=${OPENSSL_DIR}/${NDKARCH} \
  -DOPENSSL_SSL_LIBRARY=${OPENSSL_DIR}/${NDKARCH}/libssl.a \
  -DOPENSSL_CRYPTO_LIBRARY=${OPENSSL_DIR}/${NDKARCH}/libcrypto.a \
  -DLIBSSH2_LIBRARY=${LIBSSH2_DIR}/${NDKARCH} \
  -DLIBSSH2_INCLUDE_DIR=${LIBSSH2_DIR}/${NDKARCH} \
  -DBUILD_EXAMPLES=OFF \
  -DBUILD_TESTS=OFF \
  -DBUILD_SHARED_LIBS=OFF \
  -DBUILD_CLI=OFF \
  -DUSE_SSH=ON \
  -DSONAME=OFF \
  -DLINT=OFF \
  -DANDROID_ABI=${NDKARCH} \
  -DANDROID_NATIVE_API_LEVEL=${API_LEVEL} \
  -DANDROID_NDK=${NDK} \
  -DCMAKE_TOOLCHAIN_FILE=${NDK}/build/cmake/android.toolchain.cmake \
  -DCMAKE_VERBOSE_MAKEFILE=TRUE \
  -DCMAKE_BUILD_TYPE=Release \
  -DCMAKE_MAKE_PROGRAM=${MAKE}

  cd makegenerate/${NDKARCH}
  ${MAKE}
  mkdir ${OUTPUT_DIR}/${NDKARCH}
  mv liblibgit2package.a ${OUTPUT_DIR}/${NDKARCH}/libgit2.a
  cd ../..
}

#Rebuild
rm -rf makegenerate
rm -rf output/*

cp -rp ${LIBGIT2}/include output/include

#ABI x86_64
make_android x86_64

#ABI armv7a
make_android armeabi-v7a

#ABI aarch64
make_android arm64-v8a

#ABI x86
make_android x86

```

### Expected behavior
compile error.

### Actual behavior
compile error.

### Version of libgit2 (release number or SHA1)
2-1.6.4

### Operating system(s) tested
Ubuntu21.10

**Comment by boretrk** [comment]
I suspect the problem here is that the libgit2 cmake files sets the `C_STANDARD` to `C90` while using `C99` features.

I think the reason for C90 is to get some warnings for some undesired C99 features are used.
Since C99 features are used I think it would be better if libgit2 defaulted to it and used the C90 setting in ci only.
As it is now people compiling the library might end up with bunch of function calls for all the `inline static` wrappers.

**Comment by dg0yt** [comment]
Duplicates #6314

**Comment by T-Nosaka** [comment]
Thanks, eleliu008.

I see, I understand.

Will it continue to apply C90 to this project?
At least for Android, I think C99 or higher is required.

I resolved the issue by modifying the following files.
libgit2-1.7.1/src/libgit2/CMakeLists.txt:set_target_properties(libgit2 PROPERTIES C_STANDARD 90)
libgit2-1.7.1/src/util/CMakeLists.txt:set_target_properties(util PROPERTIES C_STANDARD 90)

**Comment by leleliu008** [comment]
> Will it continue to apply C90 to this project?

I'm not the member of this project. I have no idea about this.

If you don't wana track this issue, maybe use my GitHub Actions to build libgit2 is a good choice: https://github.com/leleliu008/ndk-pkg-package-manually-build

**Comment by ethomson** [comment]
Fixed via #6911 - you can now specify the `C_STANDARD` that you want to build with.
